### PR TITLE
Clean up collector configuration doc

### PR DESCRIPTION
**Pull request by cartermp**
Fixes https://github.com/open-telemetry/opentelemetry.io/issues/1899 and makes the opening statement more clear.

### CODE REVIEW

File: content/en/docs/collector/configuration.md [hunk 1]
```diff
@@ -3,12 +3,8 @@ title: "Configuration"
 weight: 20
 ---
 
-Please be sure to review the following documentation:
-
-- [Data Collection concepts](../../concepts/data-collection) in order to
-  understand the repositories applicable to the OpenTelemetry Collector.
-- [Security
-  guidance](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security.md)
```
Comment (svrnm) [inline content/en/docs/collector/configuration.md:11]: I think this file was just renamed to https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md ?

**Comment by svrnm** [comment]
Would this lead to a merge conflict for https://github.com/open-telemetry/opentelemetry.io/pull/1813/ ?

File: content/en/docs/collector/configuration.md (hunk 1 above)
Comment (cartermp) [reply content/en/docs/collector/configuration.md:11]: Oh, heh, for some reason I didn't see this as being in the collector repo rather than docs.

**Comment by cartermp** [comment]
Yep, let's just wait for #1813

**Pull request closed without merging**